### PR TITLE
BAU: Fixed redis serialisation problem

### PR DESCRIPTION
**Pull request by tomrosier**
Edited the session object to be serialised back from JSON,

Some other tidy ups along the way.

### CODE REVIEW

File: proxy-node-gateway/src/main/java/uk/gov/ida/notification/session/GatewaySessionData.java [hunk 1]
```diff
@@ -9,25 +11,51 @@
...
 
+    @NotNull
     @NotEmpty
+    @JsonProperty
     private final String eidasConnectorPublicKey;
 
+    @JsonCreator
+    public GatewaySessionData(
+        @JsonProperty("hubRequestId") String hubRequestId,
+        @JsonProperty("eidasRequestId") String eidasRequestId,
```
Comment (idrop) [inline proxy-node-gateway/src/main/java/uk/gov/ida/notification/session/GatewaySessionData.java]: Is the `@JsonProperty` annotation required on constructor args and on the field? (asking for a friend)

Comment (tomrosier) [reply proxy-node-gateway/src/main/java/uk/gov/ida/notification/session/GatewaySessionData.java]: Yeah it doesn't seem to work without it, I did try without. If anyone knows a better way id love to know 😃

Comment (CharlesIC) [reply proxy-node-gateway/src/main/java/uk/gov/ida/notification/session/GatewaySessionData.java]: There is a better way! 

You just need an empty constructor and then you don't need all that `@JsonCreator` and `@JsonProperty("...")` fluff :)

See `HubResponseTranslatorRequest` for instance

```
...
    @NotNull
    @JsonProperty
    private String connectorEncryptionCertificate;

@SuppressWarnings("Needed for JSON serialisation")
    public HubResponseTranslatorRequest() {
    }

// Normal constructor without JsonCreator fluff
public HubResponseTranslatorRequest(
            String samlResponse,
            String requestId,
            String eidasRequestId,
            String levelOfAssurance,
            URI destinationUrl,
            String connectorEncryptionCertificate) {
        this.samlResponse = samlResponse;
        this.requestId = requestId;
        this.eidasRequestId = eidasRequestId;
        this.levelOfAssurance = levelOfAssurance;
        this.destinationUrl = destinationUrl;
        this.connectorEncryptionCertificate = connectorEncryptionCertificate;
    }
...
```

Comment (tomrosier) [reply proxy-node-gateway/src/main/java/uk/gov/ida/notification/session/GatewaySessionData.java]: 😱 ❤️

**Pull request merged**